utils/calendrical_calculations/src/iso.rs

```diff
@@ -14,15 +14,15 @@ use crate::rata_die::RataDie;
 const EPOCH: RataDie = RataDie::new(1);
 
 /// Whether or not `year` is a leap year
-pub fn is_leap_year(year: i32) -> bool {
+pub const fn is_leap_year(year: i32) -> bool {
     year % 4 == 0 && (year % 400 == 0 || year % 100 != 0)
 }
 
 // Fixed is day count representation of calendars starting from Jan 1st of year 1.
 // The fixed calculations algorithms are from the Calendrical Calculations book.
 //
 /// Lisp code reference: <https://github.com/EdReingold/calendar-code2/blob/1ee51ecfaae6f856b0d7de3e36e9042100b4f424/calendar.l#L1167-L1189>
-pub fn fixed_from_iso(year: i32, month: u8, day: u8) -> RataDie {
+pub const fn fixed_from_iso(year: i32, month: u8, day: u8) -> RataDie {
```

Comment:
I've recently been sorry when we made functions `const`. It's great for things like ref-cast, but not for business logic. This function is nontrivial enough that maybe we shouldn't do it here.

Reply:
That PR actually makes `fixed_from_iso` const itself, as it uses it to define some internal constants. I generally share your concern, but the calculations for an ISO date will never need any functionality beyond the most basic arithmetic operations, which are all const.

Reply:
I dunno, you're probably right. But what's the harm in making it `const_fixed_from_iso` in case we need breathing room? I'm not convinced that the risk is low enough.

Reply:
API surface, and it will make call sites 6 characters longer, but I can do it if you insist.

Reply:
In general I am quite happy with the clean policy that:

1. If a function is _trivially_ const, it should just be const. Example: ref-cast, new from static data, generally anything that just involves pointery stuff.
2. If a function is _not trivially_ const, it should have a const version of itself. Example: this, locale parse functions, generally anything that involves business logic.

Reply:
I don't think some vague possibility of a performance optimisation that is not const is enough justification to duplicate every single one of our const functions. I have yet to see an example of a function that can be implemented in const, but can be more efficiently implemented without const.

Reply:
Maybe we cache the results for years 2000-2050 in an array or something. Random access and arrays in general are really ugly in const code, and they completely break when you add traits.

Reply:
Ugly is fine as long as it's implementable. We're not going to have two methods just so one implementation is more readable.